shapely/strtree.py

```diff
@@ -34,317 +34,185 @@ class STRtree:
-        A sequence of objects which typically serve as identifiers in an
-        application. This sequence must have the same length as geoms.
     node_capacity : int, default 10
         The maximum number of child nodes per parent node in the tree.
 
-    Examples
-    --------
-    Creating an index of polygons:
-
-    >>> from shapely.strtree import STRtree
-    >>> from shapely.geometry import Polygon
-    >>>
-    >>> polys = [Polygon(((0, 0), (1, 0), (1, 1))),
-    ...          Polygon(((0, 1), (0, 0), (1, 0))),
-    ...          Polygon(((100, 100), (101, 100), (101, 101)))]
-    >>> tree = STRtree(polys)
-    >>> query_geom = Polygon(((-1, -1), (2, 0), (2, 2), (-1, 2)))
-    >>> result = tree.query(query_geom)
-    >>> polys[0] in result
-    True
-    >>> polys[1] in result
-    True
-    >>> polys[2] in result
-    False
-
-    >>> import shapely
-    >>> tree = shapely.STRtree(shapely.points(np.arange(10), np.arange(10)))
-    >>> # Query geometries that overlap envelope of input geometries:
-    >>> tree.query(shapely.box(2, 2, 4, 4)).tolist()
-    [2, 3, 4]
-    >>> # Query geometries that are contained by input geometry:
-    >>> tree.query(shapely.box(2, 2, 4, 4), predicate='contains').tolist()
-    [3]
-    >>> # Query geometries that overlap envelopes of ``geoms``
-    >>> tree.query_bulk([shapely.box(2, 2, 4, 4), shapely.box(5, 5, 6, 6)]).tolist()
-    [[0, 0, 0, 1, 1], [2, 3, 4, 5, 6]]
-    >>> tree.nearest([shapely.points(1,1), shapely.points(3,5)]).tolist()  # doctest: +SKIP
-    [[0, 1], [1, 4]]
-
     References
     ----------
     .. [1] Leutenegger, Scott T.; Edgington, Jeffrey M.; Lopez, Mario A.
        (February 1997). "STR: A Simple and Efficient Algorithm for
        R-Tree Packing".
        https://ia600900.us.archive.org/27/items/nasa_techdoc_19970016975/19970016975.pdf
-
     """
 
     def __init__(
         self,
         geoms: Iterable[BaseGeometry],
-        items: Iterable[Any] = None,
         node_capacity: int = 10,
     ):
         # Keep references to geoms
         self.geometries = np.asarray(geoms, dtype=np.object_)
+        # prevent modification
+        self.geometries.flags.writeable = False
 
         # initialize GEOS STRtree
         self._tree = lib.STRtree(self.geometries, node_capacity)
 
-        # handle items
-        self._has_custom_items = items is not None
-        if self._has_custom_items:
-            items = np.asarray(items)
-        else:
-            # should never be accessed
-            items = None
-        self._items = items
-
     def __len__(self):
         return self._tree.count
 
     def __reduce__(self):
-        if self._has_custom_items:
-            return (STRtree, (self.geometries, self._items))
-        else:
-            return (STRtree, (self.geometries,))
+        return (STRtree, (self.geometries,))
 
-    def query_items(
-        self, geom: BaseGeometry, predicate=None, distance=None
-    ) -> Sequence[Any]:
+    def geometries(self):
```

Comment:
This is brought up to the top level so that we can provide a docstring for it; `tree.geometries` is used in the examples below, so it seemed important that this appear in the public API (which seems our overall intent with it?).